File: frontend/src/types/roast.ts
export interface RoastMetric {
  label: string
  value: string
  verdict: string
  commentary: string
}

export interface RoastEntry {
  code: string
  stock_name: string
  position_weight_pct: number
  pl_ratio: number
  roast_comment: string
}

export interface RoastSummary {
  overall_score: number
  overall_verdict: string
  intro_text: string
  metrics: RoastMetric[]
  positions_roasted: RoastEntry[]
  generated_at: string
}

File: backend/app/models/roast.py
from datetime import datetime

from pydantic import BaseModel


class RoastMetric(BaseModel):
    label: str
    value: str
    verdict: str
    commentary: str


class RoastEntry(BaseModel):
    code: str
    stock_name: str
    position_weight_pct: float
    pl_ratio: float
    roast_comment: str


class RoastSummary(BaseModel):
    overall_score: int
    overall_verdict: str
    intro_text: str
    metrics: list[RoastMetric]
    positions_roasted: list[RoastEntry]
    generated_at: datetime
